File: Projet_SHAITA_DRIOUCHE_INFO0503/serveur_PHP/package_php/enum/ModeExtractionEnergie.php
<?php

// fait une enumeration pour les mode d'extraction d'energie
class ModeExtractionEnergie {
    // attributs
    public static $AUCUNE = null;
    public static $EOLIEN = null;
    public static $NUCLEAIRE = null;
    public static $FORAGEVERTICAL = null;
    public static $FORAGEHORIZONTAL = null;
    public static $SOLAIRE = null;


    private $nom;

    // constructeur
    public function __construct($nom){
        $this->nom = $nom;
        
    }

    public function toString() : String{ return $this->nom; }

    // public static function getValue(String $value){
    //     // $value=strtoupper(Display::removeAccents($value));
    //     return self::$$value;
    // }

   
}

ModeExtractionEnergie::$AUCUNE = new ModeExtractionEnergie("Aucune");
ModeExtractionEnergie::$EOLIEN = new ModeExtractionEnergie("Eolien");
ModeExtractionEnergie::$NUCLEAIRE = new ModeExtractionEnergie("Nucleaire");
ModeExtractionEnergie::$FORAGEVERTICAL = new ModeExtractionEnergie("Forage Vertical");
ModeExtractionEnergie::$FORAGEHORIZONTAL = new ModeExtractionEnergie("Forage Horizontal");
ModeExtractionEnergie::$SOLAIRE = new ModeExtractionEnergie("Solaire");


$tabModeExtraction = array(ModeExtractionEnergie::$AUCUNE,ModeExtractionEnergie::$EOLIEN , ModeExtractionEnergie::$NUCLEAIRE , ModeExtractionEnergie::$FORAGEVERTICAL , ModeExtractionEnergie::$FORAGEHORIZONTAL , ModeExtractionEnergie::$SOLAIRE);





?>

File: Projet_SHAITA_DRIOUCHE_INFO0503/serveur_JAVA/src/package_java/enums/ModeExtractionEnergie.java
package package_java.enums;

public enum ModeExtractionEnergie {

    AUCUNE("Aucune"),
    EOLIEN("Eolien"),
    NUCLEAIRE("Nucleaire"),
    FORAGEVERTICAL("Forage Vertical"),
    FORAGEHORIZONTAL("Forage Horizontal"),
    SOLAIRE("Solaire");

    private String nom;

    ModeExtractionEnergie(String nom) {
        this.nom = nom;
    }

    public String toString() {
        return nom;
    }

    public static ModeExtractionEnergie getValue(String s) {
        for (ModeExtractionEnergie c : ModeExtractionEnergie.values()) {
            if (c.nom.equals(s))
                return c;
        }
        return null;
    }

}
